backend/src/main/java/com/javahelp/backend/domain/user/register/RegisterUserResponse.java

```diff
@@ -0,0 +1,70 @@
+package com.javahelp.backend.domain.user.register;
+
+import com.javahelp.model.user.User;
+import com.javahelp.model.user.UserPassword;
+
+public class RegisterUserResponse {
+
+    private User user;
+    private UserPassword password;
+    private String error;
+
+    /**
+     * For bundle up info of the result of this successful registration
+     * @param user the newly created {@link User}
+     * @param password the newly created {@link UserPassword}
+     */
+    public RegisterUserResponse(User user, UserPassword password){
+        this.user = user;
+        this.password = password;
+        this.error = null;
+    }
+
+    /**
+     * An overloaded constructor to make a report for this failed registration
+     * @param error a description of failure at registration
+     */
+    public RegisterUserResponse(String error){
+        this.user = null;
+        this.password = null;
+        this.error = error;
+    }
+
+    /**
+     * @return a description of failure at registering from {@link UserRegisterInteractor}.
+     * Null if succeeded; failed otherwise
+     */
+    public String getErrorMessage() {
+        return error;
+    }
+
+    /**
+     * @return an instance of a newly created {@link User}
+     * Null if failed; success otherwise
+     */
+    public User getUser(){
+        return user;
+    }
+
+    /**
+     * @return an instance of a newly made {@link UserPassword}
+     * Null if failed, success otherwise
+     */
+    public UserPassword getPassword(){
+        return password;
+    }
+
+    /**
+     * @param newUser an updated user
+     */
+    public void setUser(User newUser){
+        this.user = newUser;
+    }
+
+    /**
+     * @param newPassword an updated password
+     */
+    public void setPassword(UserPassword newPassword){
+        this.password = newPassword;
+    }
```

Comment:
This class should not have a setter for either of these fields. This is for the same reason as the constructor visibility, there is no scenario where either of these fields would be set outside of the constructor.